server/src/models/users.ts

```diff
@@ -1,26 +1,36 @@
-// const sql = require("../config/databaseHandler");
+const sql = require("../config/databaseHandler");
 
-// //Task object constructor
-// export const User = function(this: any, user: any){
-//     this.first_name = user.first_name;
-//     this.last_name = user.last_name;
-//     this.city = user.city;
-//     this.phone = user.phone;
-//     this.gender = user.gender;
-//     this.email = user.email;
-//     this.address = user.address;
-//     this.postcode = user.postcode;
-//     this.date_of_birth = user.date_of_birth;
-// };
+// Task object constructor <- I have no idea how to insert this into SQL
+export const User = function(this: any, user: any){
+    this.firebase_uid = user.firebase_uid;
+    this.first_name = user.first_name;
+    this.last_name = user.last_name;
+    this.email_addr = user.email_addr;
+    this.facebook_id = user.facebook_id;
+    this.is_anon = user.is_anon;
+    this.fcm_token = user.fcm_token;
+};
 
-// User.findAllUsers = (result: any) => {
-//     sql.query("SELECT * from user", (err: any, res: any) => {
-//         if (err) {
-//             console.log("error: ", err);
-//             result(err, null);
-//         } else {
-//             console.log(res);
-//             result(null, res);
-//         }
-//     });
-// };
\ No newline at end of file
+User.create = (newUser: any, result: any) => {
+    sql.query("INSERT INTO user SET ? ON DUPLICATE KEY UPDATE ?", [newUser, newUser], (err: any, res: any) => {
```

Comment:
Try to use the stored procedures in the database already instead of writing raw SQL in the server code. That way its easier for us to maintain the database if we decide to change things

Reply:
For example, you can do sql.query( "CALL createUser(?,?,?....)", [param1,param2,param3,...])